slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -23,24 +24,19 @@ ::grpc::Status SLAMServiceImpl::GetPosition(ServerContext *context,
                                             GetPositionResponse *response) {
     cartographer::transform::Rigid3d global_pose;
     {
-        std::lock_guard<std::mutex> lk(viam_response_mutex);
+        std::lock_guard<std::mutex> lk(latest_global_pose_mutex);
         global_pose = latest_global_pose;
     }
     // Setup mapping of pose message to the response. NOTE not using
     // inFrame->set_reference_frame
     PoseInFrame *inFrame = response->mutable_pose();
     Pose *myPose = inFrame->mutable_pose();
 
-    // set pose for our response
+    // Set pose for our response
     myPose->set_x(global_pose.translation().x());
     myPose->set_y(global_pose.translation().y());
     myPose->set_z(global_pose.translation().z());
 
-    // TODO DATA-531: Remove extraction and conversion of quaternion from the
```

Comment:
Because the [ticket](https://viam.atlassian.net/browse/DATA-531) contains the same information as this TODO comment, it describes very well that the change has to happen in this location. I don't think we should keep `TODO` comments around if they're not adding any value or if they're not serving as a placeholder for code that still has to be implemented. In this case, the comment is mostly a distraction that makes the code harder to read.